DashSync/shared/Models/Managers/Chain Managers/DSMasternodeManager.m

```diff
@@ -361,7 +363,9 @@ - (BOOL)requestMasternodeListForBlockHeight:(uint32_t)blockHeight error:(NSError
 - (BOOL)requestMasternodeListForBlockHash:(UInt256)blockHash {
     self.store.lastQueriedBlockHash = blockHash;
     NSData *blockHashData = uint256_data(blockHash);
-    [self.store.masternodeListQueriesNeedingQuorumsValidated addObject:blockHashData];
+    @synchronized (self.store.masternodeListQueriesNeedingQuorumsValidated) {
```

Comment:
@pankcuf I don't think @synchronized is a real solution, I personally against using @synchronized because it can lead performance decreasing and other "unknowns"